lib/system/dollars.nim

```diff
@@ -160,3 +165,35 @@ proc `$`*[T](x: openArray[T]): string =
   ## .. code-block:: Nim
   ##   $(@[23, 45].toOpenArray(0, 1)) == "[23, 45]"
   collectionToString(x, "[", ", ", "]")
+
+proc `$`*[T: ref | ptr](arg: T): string =
+  if arg == nil:
+    "nil"
+  else:
+    $arg[]
+
+proc `$`*(arg: pointer): string =
+  if arg == nil:
+    return "nil"
+  else:
+    # maybe print something so people know it is a pointer.
+    result.add "0x"
+    const
+      HexChars = "0123456789ABCDEF"
+    let len = sizeof(pointer)
+    let arg = cast[uint](arg)
+    for i in countdown(sizeof(pointer)-1, 0):
+      result.add HexChars[(arg shr i) and 0xf]
+
+proc distinctBase(T: typedesc): typedesc {.magic: "TypeTrait".}
+
+proc `$`*[T: distinct](arg: T): string =
```

Comment:
Desaster for my ``distinct string`` type that represents SQL queries and is distinct in order to fight SQL injection attacks.

Reply:
Well, I will make an RFC about the "all types can be printed". So you don't need to answer here right now. But I disagree on the "desaster" part here. After all this is the `dollar` operator, not an implicit conversion. This is the operator that gets applied to all arguments of ``echo`` and other ``varargs[string, `$`]`` procs that explicitly require this type of string conversion.

Reply:
I tried to cover your concerns in my RFC here: https://github.com/nim-lang/RFCs/issues/203
But please elaborate it a bit more. Best would be an example where you rely on a compiler error message that would be absent in my change.